src/tagging/components/InnerComponents/ValueSelector.js

```diff
@@ -4,53 +4,98 @@ import PropTypes from 'prop-types';
 import { __ } from '../../../global-functions';
 import TaggingPropTypes from '../TaggingPropTypes';
 import customStyles from '../../../forms/select-styles';
+import { ValueContainer } from '../../../forms/pf-select';
+import '../../sass/multiSelectFix.scss';
 
 class ValueSelector extends React.Component {
-  getvalues = values =>
+  getValues = values =>
     values.map(tag => ({ value: tag.id, label: tag.description }));
 
-  handleChange = (selectedOption) => {
-    this.props.onTagValueChange({
-      description: selectedOption.label,
-      id: selectedOption.value,
-    });
+  /**
+   * Maps the selected options for the select box input.
+   *
+   */
+  getActiveValues = () => {
+    if (this.props.selectedOption.length > 0) {
+      return this.props.selectedOption.map(el => ({ value: el.id, label: el.description }));
+    }
+    return [];
+  };
+
+  /**
+   * Handles the single and multi select value changes.
+   *
+   * Sends an Array of {description, id} to the callback.
+   * If null is received sends and empty array instead
+   * (happens when removing elements one by one).
+   * On reset button (multi select) sends the default empty array.
+   *
+   * @param values {label, value}[] Currently selected values.
+   */
+  handleChange = (values) => {
+    // Deleting all the multi selected items
+    if (values === null) {
+      return this.props.onTagValueChange([]);
+    }
+
+    // Multi select
+    if (Array.isArray(values)) {
```

Comment:
**Nitpick**: You can simplify this by adding inline `isArray` check and check if values exist. It looks more complicaed this way, so if you want you can change just the second part (check if values is array and if not true create new array with one item `values`)
```JS
const tagValues = values ? (Array.isArray(values) ? values : [values]) : [];
return this.props.onTagValueChange(tagValues.map(el => ({
  description: el.label,
  id: el.value,
}))
```

Reply:
Nice, but I let the first check intact since linter doesn't like nested ternary operators.

Reply:
Yeah, sure. Just change the second one so it is simpler
```
return this.props.onTagValueChange((Array.isArray(values) ? values : [values]).map(el => ({
  description: el.label,
  id: el.value,
}))
```